x509/src/extensions_utils.rs

```diff
@@ -0,0 +1,142 @@
+//! Utility functions and enums related to X.509 extensions
+
+use alloc::vec::Vec;
+use core::fmt;
+use der::asn1::BitString;
+
+/// Enum representing values from the KeyUsage structure
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum KeyUsageValues {
+    /// DigitalSignature
+    DigitalSignature = 0,
+    /// NonRepudiation
+    NonRepudiation = 1,
+    /// KeyEncipherment
+    KeyEncipherment = 2,
+    /// DataEncipherment
+    DataEncipherment = 3,
+    /// KeyAgreement
+    KeyAgreement = 4,
+    /// KeyCertSign
+    KeyCertSign = 5,
+    /// CRLSign
+    CRLSign = 6,
+    /// EncipherOnly
+    EncipherOnly = 7,
+    /// DecipherOnly
+    DecipherOnly = 8,
+}
+
+impl fmt::Display for KeyUsageValues {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        match *self {
+            KeyUsageValues::DigitalSignature => write!(f, "DigitalSignature"),
+            KeyUsageValues::NonRepudiation => write!(f, "NonRepudiation"),
+            KeyUsageValues::KeyEncipherment => write!(f, "KeyEncipherment"),
+            KeyUsageValues::DataEncipherment => write!(f, "DataEncipherment"),
+            KeyUsageValues::KeyAgreement => write!(f, "KeyAgreement"),
+            KeyUsageValues::KeyCertSign => write!(f, "KeyCertSign"),
+            KeyUsageValues::CRLSign => write!(f, "CRLSign"),
+            KeyUsageValues::EncipherOnly => write!(f, "EncipherOnly"),
+            KeyUsageValues::DecipherOnly => write!(f, "DecipherOnly"),
+        }
```

Comment:
For something like this where there's a 1:1 mapping of variants to strings, I often find it helpful to have an `fn as_str(self) -> &'static str` method providing that mapping.

If you have that, the body of `Display::fmt` can just be:

```suggestion
        f.write_str(self.as_str())
```

Reply:
Done.